### ISSUE TITLE
seed data and app view

### ISSUE THREAD

**Issue by kevinhughes27** [kevinhughes27/tournament-app#6]

**Comment by theoandtheb** [comment]
I'll try and get this worked out tonight if I can.  I may try and get a hold of Chris to see if he has a spreadsheet displaying the games currently scheduled for No Borders.  I can always cheat and just make some up, but if he has this sorted, there's no reason not to base our seed on the data he has put together thus far.

Fields are obviously in flux at the moment, as I do not see a column set aside for storing the coordinates needed to build these linestrings/polygons yet.  Not sure what the plan is for that.  I could work with @vanessaj on setting up RGeo if needed (I need to sort out a buildpack for the required binaries for another project anyway).  It does a fine job of building storable shapefiles which can be easily outputted in a variety of formats.  I'd really like to know what you two have in mind for saving/storing/displaying the field shapes.

**Comment by kevinhughes27** [comment]
I have the fields stuff under control should have a PR soon (https://github.com/kevinhughes27/tournament-app/tree/field-creator). I don't think we'll need anything like RGeo. I envision our app being a pretty simple backend with all the niceties coming from javascript in the views. 

Yes it would be good to get the schedule spreadsheet from last year!

**Comment by theoandtheb** [comment]
I'd like to avoid using RGeo as well, as it is a lot of work to setup for
little payoff, however I cannot envision how we will sensibly store the
coordinates defining the linestring without using RGeo or Geokit.  I
believe we could store them as an array of decimals in two matching columns
(one for lat, one for long) and create methods for parsing the hash
generated by Google Maps and for reproducing the hash needed to reproduce
the shapes.

If there were anything else we might use RGeo for, I'd highly recommend
using the PostGIS to store the coordinates and OGR for transcription both
ways.

**Comment by kevinhughes27** [comment]
I've got the data from google maps as a set of points so I was just going to add a bunch of db columns. I can't see anything overly wrong with this approach atm. I can re-draw on google maps with just the 4 points

**Comment by theoandtheb** [comment]
The problem I foresee is what prevents a user from defining a field with
only four points?  Odds are the majority will do just that (it only makes
sense), but what will happen if someone misclicks and accidentally adds a
fifth point?  What if they choose to define the area outside of the field
(a spectator's gallery for example)?

The lat and long decimal columns set to array: true would allow for the
storage of and iteration through of any number of points.  That, of course,
assumes that ActiveRecord and Postgres both support the decimal datatype
with array: true (I couldn't find an example the last time I checked,
though I haven't sampled GutHub-hosted migrations or schema's for a similar
use case yet).

Just a thought.

**Comment by kevinhughes27** [comment]
I added code to the map drawing that validates that we only have 4 points in my branch

**Comment by theoandtheb** [comment]
I see.  So long as I know what to seed for our fields, I'll generate
content for the seeds to conform to that standard.  Will wait until that
branch has been merged before creating that part of the seed file.

**Comment by kevinhughes27** [comment]
:+1: I'll try and get it out tonight

**Comment by theoandtheb** [comment]
Great.  I'll ask Chis via Slack for his games spreadsheet and start
hammering away at seeding the No Borders tourney.  I may need to ask him
about the spirit sheets, as I'd like to know if they should persist beyond
a given tournament.  Probably beyond the scope of this project, as teams do
not persist beyond a given tournament, but if the same spirit sheets are
used for all tournaments and teams participate in multiple tournaments,
there would be value in carrying forward those results from one tournament
to the next.

**Comment by mmasse** [comment]
Spirit should have the same persistance as the tournament/games
And should not spill over to other tournaments.

In my mind teams persisting past a tournament is up for debate, as they
will be entered individually for each tournament.
But I suppose it would be nice to have the ability to see how a team
performs in multiple tournaments, linking the same team between tournaments
might be a pain.

M

**Comment by mmasse** [comment]
Not even sure if email replies to github updates work.....

**Comment by theoandtheb** [comment]
Matt:  I don't know if they show up on GitHub, but Kevin and I have been
messaging back and forth throughout the day by replying to these
GutHub-generated emails.

**Comment by theoandtheb** [comment]
I'm sure there is a lot of fluidity from tournament to tournament that
persistence between tournaments makes little sense.

Until we hear otherwise from Chris and Kevin, I'll assume no persistence.

**Comment by mmasse** [comment]
email replies work, I am also getting your back and forth discussions :-)

**Comment by mmasse** [comment]
Which is fine, I will try and help out too.

**Comment by kevinhughes27** [comment]
yeah github will take the email and post if here which is pretty cool. If you don't want emails about a repo then you can unsubscribe at the top. You can also unsubscribe from a given issue as well.

I think the data model we want is keep everything scoped to a tournament but make it easy to export and then import to a new tournament to save on data entry.
